src/models/model.cpp

```diff
@@ -1879,6 +1895,12 @@ void model::remove_layer(std::string const& removable_layer_name) {
   child.replace_parent_layer(l.get_parent_layer_pointer(0), child.find_parent_layer_index(l));
   parent.replace_child_layer(l.get_child_layer_pointer(0), parent.find_child_layer_index(l));
 
+  // Remove weights for the old layer
+  auto old_weights_ptrs = m_layers[removable_layer_index]->get_weights_pointers();
+  for (const auto w : old_weights_ptrs) {
+     this->remove_weights(std::shared_ptr<weights>(w)->get_name());
+  }
```

Comment:
This behavior will cause problems if the layer is sharing weights. It's fine since that's not our current use-case, but it would be good to mention that behavior in the documentation.